File: SDKs/files.com/0.0.1/typescript/src/sdk/models/operations/folderlistforpath.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class FolderListForPathPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=path" })
  path: string;
}


export class FolderListForPathQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=filter" })
  filter?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=preview_size" })
  previewSize?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=search" })
  search?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=search_all" })
  searchAll?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=with_previews" })
  withPreviews?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=with_priority_color" })
  withPriorityColor?: boolean;
}


export class FolderListForPathRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: FolderListForPathPathParams;

  @SpeakeasyMetadata()
  queryParams: FolderListForPathQueryParams;
}


export class FolderListForPathResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata({ elemType: shared.FileEntity })
  fileEntities?: shared.FileEntity[];

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/files.com/0.0.1/python/src/sdk/models/operations/folderlistforpath.py
from dataclasses import dataclass, field
from typing import List,Optional
from sdk.models import shared


@dataclass
class FolderListForPathPathParams:
    path: str = field(metadata={'path_param': { 'field_name': 'path', 'style': 'simple', 'explode': False }})
    

@dataclass
class FolderListForPathQueryParams:
    cursor: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'cursor', 'style': 'form', 'explode': True }})
    filter: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'filter', 'style': 'form', 'explode': True }})
    per_page: Optional[int] = field(default=None, metadata={'query_param': { 'field_name': 'per_page', 'style': 'form', 'explode': True }})
    preview_size: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'preview_size', 'style': 'form', 'explode': True }})
    search: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'search', 'style': 'form', 'explode': True }})
    search_all: Optional[bool] = field(default=None, metadata={'query_param': { 'field_name': 'search_all', 'style': 'form', 'explode': True }})
    with_previews: Optional[bool] = field(default=None, metadata={'query_param': { 'field_name': 'with_previews', 'style': 'form', 'explode': True }})
    with_priority_color: Optional[bool] = field(default=None, metadata={'query_param': { 'field_name': 'with_priority_color', 'style': 'form', 'explode': True }})
    

@dataclass
class FolderListForPathRequest:
    path_params: FolderListForPathPathParams = field()
    query_params: FolderListForPathQueryParams = field()
    

@dataclass
class FolderListForPathResponse:
    content_type: str = field()
    status_code: int = field()
    file_entities: Optional[List[shared.FileEntity]] = field(default=None)
    
